### PR TITLE
Add auto assign action

### PR DESCRIPTION
**Pull request by nirinchev**
When a PR is created or moved to Open (i.e. no longer a draft), this action will request a review from everyone and assign the author of the PR as assignee.

Example run: https://github.com/realm/realm-dart/actions/runs/3192205881/jobs/5209419299

Fixes https://github.com/realm/realm-dart/issues/941

### CODE REVIEW

**Review by nielsenko — APPROVED** [review]
LGTM

**Review by desistefanova — COMMENTED** [review]
Will this request reviews if I create a draft PR?

**Comment by nirinchev** [comment]
No, it only requests reviews if a draft PR is moved to "Open" or if you create a non-draft PR.

**Review by desistefanova — APPROVED** [review]
This means that the assignments also will be added only for opened for review PRs. We will have to assign our Draft PRs manually. But I don't think it is a problem.

**Comment by nirinchev** [comment]
Yes, for draft PRs, you typically don't want to assign the entire team - just 1-2 people who you want early feedback from.

**Review by blagoev — CHANGES_REQUESTED** [review]
Lets discuss this again before merging. (I added a topic.) Right now this will request a review immediately when a PR is opened and requests can not be undone. 
Otherwise the change is alright.

**Review by blagoev — APPROVED** [review]

**Review by desistefanova — APPROVED** [review]

File: .github/workflows/auto-assign.yml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+name: 'Auto Assign'
+on:
+  pull_request:
+    types: [opened]
+
+jobs:
+  add-reviews:
```
Comment (desistefanova) [inline .github/workflows/auto-assign.yml:7]: I forgot to mention that `add-reviews:` now should ne `add-assignee:`. I can change it in my PR together with updating samples workflow.

Comment (nirinchev) [reply .github/workflows/auto-assign.yml:7]: Yeah the name doesn't really matter, but you're right that it can be more precise - feel free to change it.

**Pull request merged**